lua/heirline/statusline.lua

```diff
@@ -272,32 +272,56 @@ local function register_global_function(component)
     return "v:lua." .. func_name
 end
 
+---@param data table
+local function create_event(data)
+    local events = {}
+    for _, e in ipairs(data) do
+        if type(e) == "string" then
+            tbl_insert(events, e)
+        end
+    end
+    return {
+        events = events,
+        callback = data.callback,
+        pattern = data.pattern,
+    }
+end
+
 ---@param component StatusLine
 local function register_update_autocmd(component)
-    local events, callback, pattern
+    local events = {}
+
     if type(component.update) == "string" then
-        events = component.update
+        tbl_insert(events, {
+            event = component.update,
+        })
     else
-        events = {}
-        for i, e in ipairs(component.update) do
-            tbl_insert(events, e)
+        local event = create_event(component.update)
+        if #event.events > 0 then
+            tbl_insert(events, event)
         end
-        callback = component.update.callback
-        pattern = component.update.pattern
-    end
 
-    local id = vim.api.nvim_create_autocmd(events, {
-        pattern = pattern,
-        callback = function(args)
-            component._win_cache = nil
-            if callback then
-                callback(component, args)
+        for _, e in ipairs(component.update) do
+            if type(e) == "table" then
+                tbl_insert(events, create_event(e))
             end
-        end,
-        desc = "Heirline update autocmd for " .. vim.inspect(component.id),
-        group = "Heirline_update_autocmds",
-    })
-    component._au_id = id
+        end
+    end
+
+    for _, e in ipairs(events) do
+        vim.api.nvim_create_autocmd(e.events, {
+            pattern = e.pattern,
+            callback = function(args)
+                component._win_cache = nil
+                if e.callback then
+                    e.callback(component, args)
+                end
+            end,
+            desc = "Heirline update autocmd for " .. vim.inspect(component.id),
```

Comment:
Would it be helpful to include something unique to each event (`e`) in the `desc`ription here?

Maybe the events and possibly the pattern?

Something like:
```lua
desc = "Heirline update autocmd for " .. vim.inspect(component.id) .. " " .. table.concat(events, ", ") .. " " .. pattern,
```